### PR TITLE
Detail Screens (Including Tabs) in CLI

### PR DESCRIPTION
**Pull request by ctsims**
Added compound screen mechanism to provide navigation between internal states, and implemented both single (no-tab) and compound (multiple tab) entity detail screens

### CODE REVIEW

File: util/src/org/commcare/util/cli/ApplicationHost.java [hunk 1]
```diff
@@ -106,14 +103,22 @@ private void processAppUpdate() {
...
+                }
+
                 System.out.println("\n\n\n\n\n\n");
+                System.out.println(s.getWrappedDisplaytitle(mSandbox, mPlatform));
+
+                System.out.println("====================");
                 s.prompt(System.out);
                 System.out.print("> ");
 
+                screenIsRedrawing = false;
```
Comment (orangejenny) [inline util/src/org/commcare/util/cli/ApplicationHost.java]: this line doesn't look necessary

File: util/src/org/commcare/util/cli/EntityListSubscreen.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        for (DetailField field : fields) {
+            Object o = field.getTemplate().evaluate(context);
+            String s;
+            if (!(o instanceof String)) {
+                s = "";
+            } else {
+                s = (String)o;
+            }
+
+            int widthHint = SCREEN_WIDTH / fields.length;
```
Comment (orangejenny) [inline util/src/org/commcare/util/cli/EntityListSubscreen.java]: Non-blocking (and I realize you're just moving this, I just haven't looked at much of the CLI code yet) but I'm surprised you're bothering with template width hints at all, doesn't seem like they'd ever translate well to an ASCII interface.

Comment (ctsims) [inline util/src/org/commcare/util/cli/EntityListSubscreen.java]: They actually look surprisingly good! we have 100 columns for characters,
which actually might be more than on a lot of phones :p.

Comment (orangejenny) [inline util/src/org/commcare/util/cli/EntityListSubscreen.java]: oh, that actually makes a ton of sense - the percentages make sense, then.

**Pull request merged**